src/utils/calculateReClammInitAmounts.ts

```diff
@@ -1,80 +1,94 @@
-        initialTargetPrice,
-    );
+	const proportion = computeInitialBalanceRatio(
+		initialMinPrice,
+		initialMaxPrice,
+		initialTargetPrice,
+	);
 
-    // Create token instances for decimal scaling helpers
-    const givenToken = new Token(
-        chainId,
-        givenAmountIn.address,
-        givenAmountIn.decimals,
-    );
-    const calculatedToken = new Token(
-        chainId,
-        sortedTokens[calculatedTokenIndex].address,
-        sortedTokens[calculatedTokenIndex].decimals,
-    );
+	// Create token instances for decimal scaling helpers
+	const givenToken = new Token(
+		chainId,
+		givenAmountIn.address,
+		givenAmountIn.decimals,
+		undefined,
+		undefined,
+		undefined,
+		sortedTokens[givenTokenIndex]?.rate,
+	);
+	const calculatedToken = new Token(
+		chainId,
+		sortedTokens[calculatedTokenIndex].address,
+		sortedTokens[calculatedTokenIndex].decimals,
+		undefined, // symbol
+		undefined, // name
+		undefined, // wrapped
+		sortedTokens[calculatedTokenIndex]?.rate, // rate
+	);
 
-    // Scale given amount to 18 decimals before doing math
-    const givenTokenAmountScaled18 = TokenAmount.fromRawAmount(
-        givenToken,
-        givenAmountIn.rawAmount,
-    ).scale18;
+	// Scale given amount to 18 decimals before doing math
+	const givenTokenAmount = givenToken.rate
+		? TokenAmountWithRate.fromRawAmountWithRate(
+				givenToken,
+				givenAmountIn.rawAmount,
+				givenToken.rate,
+		  )
+		: TokenAmount.fromRawAmount(givenToken, givenAmountIn.rawAmount);
 
-    // Do math based on token order: https://github.com/balancer/reclamm/blob/8207b33c1ab76de3c42b015bab5210a8436376de/test/reClammPool.test.ts#L120-L128
-    const rawAmountScaled18 =
-        givenTokenIndex === 0
-            ? MathSol.mulDownFixed(givenTokenAmountScaled18, proportion) // if given token is first in sort order, we multiply
-            : MathSol.divDownFixed(givenTokenAmountScaled18, proportion); // if given token is second in sort order, we divide
+	// Do math based on token order: https://github.com/balancer/reclamm/blob/8207b33c1ab76de3c42b015bab5210a8436376de/test/reClammPool.test.ts#L120-L128
+	const rawAmountScaled18 =
+		givenTokenIndex === 0
+			? MathSol.mulDownFixed(givenTokenAmount.scale18, proportion) // if given token is first in sort order, we multiply
+			: MathSol.divDownFixed(givenTokenAmount.scale18, proportion); // if given token is second in sort order, we divide'
 
-    // Convert back to native token decimals of the calculated token (rounding down)
-    const rawAmountScaledNative = TokenAmount.fromScale18Amount(
-        calculatedToken,
-        rawAmountScaled18,
-    );
+	// Convert back to native token decimals of the calculated token (rounding down)
+	const calculatedTokenScaledNative = calculatedToken.rate
+		? TokenAmountWithRate.fromScale18AmountWithRate(
+				calculatedToken,
+				rawAmountScaled18,
+				calculatedToken.rate,
+		  )
+		: TokenAmount.fromScale18Amount(calculatedToken, rawAmountScaled18);
```

Comment:
For the failling test `'6 decimal token with rate given in'`

The amount being calculated for WETH is too large. maybe this is spot that needs fixing. 

played around with only using `TokenAmountWithRate` and passing a default value for rate. this experiment fixes the one failing test but breaks others 

```
	const calculatedTokenScaledNative =
		TokenAmountWithRate.fromScale18AmountWithRate(
			calculatedToken,
			rawAmountScaled18,
			calculatedToken.rate ?? WAD * WAD,
		);
```